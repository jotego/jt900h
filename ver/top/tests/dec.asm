    ; Arithmetic operations that read & write to memory
    main section code
    org 0
    ld a,0xbf    ; common header

    ld a,7
    dec 3,a
    cp a,4
    jp ne,bad_end

    dec 1,a
    cp a,3
    jp ne,bad_end

    lda xix,data
    ld b,(xix)
    dec 3,(xix)
    ld c,(xix)
    sub b,c
    cp b,3
    jp ne,bad_end

end_loop:
    ld hl,0xbabe
    ld (0xffff),0xff
    jp end_loop
bad_end:
    ld hl,0xdead
    ld (0xffff),0xff
    jp bad_end
data:
    dw 0xcafe,0xbeef,0xffff,0xeeee,0xcccc
    end