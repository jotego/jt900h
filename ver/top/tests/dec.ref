00000003 - 00000000 - 00000000 - 00000000
000003fb - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
0000babe - 00000000 - 00000000 - 00000000
-- index registers --
00000042 - 00000000 - 00000000 - 00000100
SR = f842
