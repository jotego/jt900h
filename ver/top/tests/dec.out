00000003 - 00000000 - 00000000 - 00000000
000003fb - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
0000babe - 00000000 - 00000000 - 00000000
-- index registers --
0000003d - 00000000 - 00000000 - 00000000
SR = f842
